wellness.sql

```diff
@@ -104,18 +104,137 @@ aggregate_wellness_survey_data AS (
     GROUP BY
         site_short
 ),
---Growth KPIs and students completing Covitality KPI
-prep_all_wellness_kpis AS (
+--Average # of sessions for reb/blue Covi students or 1:1 (case notes)
+--% of students/# of sessions/amt of time that students with red and blue CoVi scores have spent receiving support/counseling/coaching for their social emotional wellbeing health (either through a workshop, small group or 1:1s)
+--Broken down into various CTEs to gather students with red/blue covi, wellness sessions attended, and case notes logged
+--Pulling students with red/blue covi from academic term (2020-21AY)
+gather_red_blue_covi_at AS (
     SELECT
-        A.site_short,
-        wellness_covi_assessment_completed_ay,
-        wellness_survey_wellness_services_assisted_num,
-        wellness_survey_wellness_services_assisted_denom
+        student_c,
+        site_short,
+        CASE
+            WHEN co_vitality_scorecard_color_c IN ('Blue', 'Red') THEN 1
+            ELSE NULL
+        END AS wellness_blue_red_denom
+    FROM
+        `data-warehouse-289815.salesforce_clean.contact_at_template`
+    WHERE
+        grade_c != '8th Grade'
+        AND college_track_status_c = '11A'
+        AND AY_NAME = "AY 2020-21"
+    GROUP BY
+        student_c,
+        site_short,
+        co_vitality_scorecard_color_c
```

Comment:
@isabelreyna, I think because the field co_vitality_scorecard_color_c comes from the academic term you could be accidentally double-counting students because of this groupby statement. 

If a student has two covi scores, one red and one blue, in a given AY then they will show up twice in this query (one for each row). Then when you sum wellness_blue_red_denom you will get "2" in the denominator for what is only one student.

Reply:
@jbakerr That is a good point.  I validated the outcomes with Salesforce and it seemed right, but this was an issue I was thinking about while building this because the Covi Indicator is not on the contact. Need to check if I meant to keep the group by student_c because I was doing that initially to validate some data

Need to confirm from a KPI perspective, with Wellness, also, which students to look at.  If a student took Covi assessment in the Spring and had a Red/Blue indicator, but were not Blue/Red before, I don't think they should be included since staff may not have had a chance to deliver the appropriate Wellness support by then to be included in an avg.

Reply:
@isabelreyna Apologies if i'm misunderstanding the issue. But couldn't you also add a Max( before the red blue case statement. Then when you group by student you'd only get 1 or 0 for each student row

Reply:
@jvolan123 that may solve it!

Reply:
@isabelreyna to apply MAX to case statement to pull in 1 row

Reply:
change made